angular-primeng-app/src/app/app.component.ts

```diff
@@ -23,6 +24,9 @@ export class AppComponent implements OnInit, OnDestroy {
 	themeService: ThemeService = inject(ThemeService);
 	blogService: BlogService = inject(BlogService);
 	private querySubscription?: Subscription;
+	private scrollEvntSub?: Subscription;
+	enableScrollUp: boolean=false;
```

Comment:
@DongreJaipal for over here, I am not sure that we need this to be a subscription. 
so let's change `private scrollEvntSub?: Subscription;` to be - `showScrollButton: boolean = false;`